HW4_Trofimov/das_protein_tools.py

```diff
@@ -0,0 +1,361 @@
+        return mass
+    else:
+        raise ValueError("Sequence is not a protein, input should be protein")
+
+
+# Function to translate Protein to RNA
+def translate_protein_rna(seq: str) -> str:
+    """
+    This function takes  aminoacid sequence and translates in to the RNA.
+    As most of the aminoacids are coded with several different codons,
+    this function will take a random codon of the set for such aminoacids.
+
+    Arguments:
+        seq (str): A sequence of RNA molecule
+
+    Output:
+        returns sequence of aminoacids
+    """
+    seq = seq.upper()
+    if is_protein(seq) is True:
+        rna = ""
+        for aa in seq:
+            codon = choice(pd.aa_codon_dict.get(aa))
+            rna += codon
+        return rna
+    else:
+        raise ValueError("Sequence is not a protein, input should be a protein")
+
+
+def main(*args: str):
+    """
+    Main function to perform various actions on protein sequences.
+
+    Args:
+    - *args: Variable number of arguments. The first n-1 arguments should be protein sequences,
+             and the last argument should be a string specifying the action to be performed.
+
+    Returns:
+    - The result of the specified action on the input protein sequences.
+
+    Raises:
+    - ValueError: If the specified action is not supported or if there is an error in the number of sequences.
+                  Also raised if the input sequences are not valid protein sequences.
+
+    Supported Actions:
+    - "get_pI": Calculate isoelectric points for each amino acid in the sequence.
+    - "needleman_wunsch": Perform global alignment of two sequences using the Needleman-Wunsch algorithm.
+    - "build_scoring_matrix": Build a scoring matrix for amino acid pairs.
+    - "calculate_aa_freq": Calculate the frequency of each amino acid in a protein sequence.
+    - "translate_protein_rna": Translate amino acid sequence to RNA, using random codons for each amino acid.
+    - "convert_to_3L_code": Convert one-letter amino acid sequence to three-letter coding.
+    - "protein_mass": Calculate the molecular weight of the protein sequence.
+    """
+
+    action = args[-1]
+    action_list = {
+        "get_pI": get_pI,
+        "needleman_wunsch": needleman_wunsch,
+        "build_scoring_matrix": build_scoring_matrix,
+        "calculate_aa_freq": calculate_aa_freq,
+        "translate_protein_rna": translate_protein_rna,
+        "convert_to_3L_code": convert_to_3L_code,
+        "protein_mass": protein_mass,
+    }
+
+    if action not in action_list:
+        raise ValueError(f"No such action: {action}")
+
+    if not (
+        action == "needleman_wunsch"
+        and len(args) == 3
+        or action != "needleman_wunsch"
+        and len(args) == 2
+    ):
+        raise ValueError("Error in number of sequences")
+
+    for sequence in args[:-1]:
+        if not all([letter.capitalize() in AMINO_LETTERS for letter in sequence]):
```

Comment:
Ой, то есть мы итерируемся для каждого сиквенса по каждой его аминокислоте и по всем возможным аминокислотам ?(( Долго и неоптимально
Множествами это сделать было бы быстрее
```suggestion
        if not set(sequence).issubset(set(AMINO_LETTERS)):
```